src/windows/cheats.c

```diff
@@ -499,6 +565,81 @@ static void window_cheats_misc_dropdown(rct_window *w, int widgetIndex, int drop
 	game_do_command(0, GAME_COMMAND_FLAG_APPLY, CHEAT_FORCEWEATHER, dropdownIndex, GAME_COMMAND_CHEAT, 0, 0);
 }
 
+enum
+{
+	CONVERT_NUMBER_TO_STRING,
+	CONVERT_STRING_TO_NUMBER,
+};
+
+static void window_cheats_convert_money(int convertMethod, char* text) 
+{
+	const char* strDecSym = language_get_string(STR_LOCALE_DECIMAL_POINT);
+	if (convertMethod == CONVERT_NUMBER_TO_STRING) {
+		if (money_spinner_value % 10 > 0) {
+			snprintf(money_spinner_text, MONEY_DIGITS, "%d%s%d0", money_spinner_value / 10, strDecSym, money_spinner_value % 10);
+		} else {
+			snprintf(money_spinner_text, MONEY_DIGITS, "%d", money_spinner_value / 10);
+		}
+	}
+	else if (convertMethod == CONVERT_STRING_TO_NUMBER) {
+		char * text_ptr = text;
+		int i, j;
+		//Remove everything except numbers and decimal
+		for (i = 0; text_ptr[i] != '\0'; ++i) {
+			while (!((text_ptr[i] >= '0' && text_ptr[i] <= '9') || 
+					(text_ptr[i] == strDecSym[0]) || 
+					(text_ptr[i] == '\0'))
+					) {
+					//move everything over to the left by one
+					for (j = i; text_ptr[j] != '\0'; ++j) {
+						text_ptr[j] = text_ptr[j + 1];
+					}
+					text_ptr[j] = '\0';
+			}
+		}
+		//determine if decimal exists in text
+		char *decimal_place = strstr(text, strDecSym);
+		if (decimal_place == NULL) {
+			//if period does not exist, keep it basic. multiply by 10 to fill decimal and be done.
+			money_spinner_value = atoi(text) * 10;
+		}
+		else {
+			char *end;
+			//Make a copy of the text to remove the decimal and see if it's zero.
+			char *strNoDec = _strdup(text);
+			str_remove_substring(strNoDec, strDecSym);
+			double val = strtol(strNoDec, &end, 10);
+			if (*end == '\0') {
+				//if value is zero, set it and be done
+				if (val == 0) {
+					money_spinner_value = 0;
+				}
+				else {
+					//value is not zero, so reparse text and split by decimal
+					val = strtol(text, &end, 10);
+					if (*end != '\0') {
+						char *tokenize = _strdup(text);
+						char *numberText = strtok(tokenize, strDecSym);
+						char *decimalText = strtok(NULL, strDecSym);
+
+						int number = 0, decimal = 0;
+						if (numberText != NULL) number = atoi(numberText);
+						if (decimalText != NULL) decimal = atoi(decimalText);
+						if (decimal < 10) decimal *= 10; //if less than 10, say, 6, convert to 60.
+
+						money_spinner_value = MONEY(number, decimal);
+
+						free(tokenize);
+						free(numberText);
```

Comment:
As far as I know, you must free things you allocated via malloc (and related functions). If you didn't use malloc, you mustn't free.